### PR TITLE
log backup: reset store grpc connection when encounter error

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/50445
Problem Summary:
advancer will cache the store grpc connection at first time connect to TiKV. This usually works well when TiKV normal scale-out/in or deploy with a domain name. but in some cases. when deploy TiKV cluster with IP address and IP changed. 
the cached grpc connection will broken.

### What changed and how does it work?
Disable the cached grpc connection when encounter error.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue that TiKV IP address changed on same store makes log backup stuck.
```

### CODE REVIEW

File: br/pkg/utils/store_manager.go [hunk 1]
```diff
@@ -192,26 +212,14 @@ func (mgr *StoreManager) WithConn(ctx context.Context, storeID uint64, f func(*g
...
 	var (
 		conn *grpc.ClientConn
 		err  error
 	)
+	err = mgr.RemoveConn(ctx, storeID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
 	for retry := 0; retry < resetRetryTimes; retry++ {
 		conn, err = mgr.getGrpcConnLocked(ctx, storeID)
```
Comment (Leavrth) [inline br/pkg/utils/store_manager.go:224]: No locked Since move line 199 out

Comment (3pointer) [reply br/pkg/utils/store_manager.go:228]: fixed, PTAL again~

**Review by Leavrth — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**